code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,183 @@
+	dat += "You have earned a total of <td>[round(GBPearned)]</td> this shift<br>"
+	dat += "</div>"
+	dat += "<br><b>Equipment point cost list:</b><BR><table border='0' width='300'>"
+	for(var/datum/GBP_equipment/prize in prize_list)
+		dat += "<tr><td>[prize.equipment_name]</td><td>[prize.cost]</td><td><A href='?src=\ref[src];purchase=\ref[prize]'>Purchase</A></td></tr>"
+	dat += "</table>"
+
+	var/datum/browser/popup = new(user, "vending", "Engineering Point Redemption", 400, 350)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/engi_points_manager/Topic(href, href_list)
+	if(..())
+		return
+	if(href_list["purchase"])
+		var/datum/GBP_equipment/prize = locate(href_list["purchase"])
+		if (!prize || !(prize in prize_list))
+			return
+		if(prize.cost > GBP)
+			return
+		else if(prize.cost <= GBP) // Placeholder spaghetti calm your shit
+			GBP -= prize.cost
+			for(var/i in 1 to prize.amount)
+				for(var/obj/machinery/engi_points_delivery/D in machines)
+					D.icon_state = "geardist-load"
+					playsound(D, 'sound/machines/Ding.ogg', 100, 1)
+					spawn(20)
+						new prize.equipment_path(get_turf(D))
+						D.icon_state = "geardist"
+					if(prize.cost == 20000) // Still a placeholder
+						new /obj/item/clothing/head/helmet/space/chronos(get_turf(src))
+					feedback_add_details("Engi_equipment_bought",
+					"[src.type]|[prize.equipment_path]")
+		else
+			GBP -= prize.cost
+			new prize.equipment_path(src.loc)
+			feedback_add_details("Engi_equipment_bought",
+				"[src.type]|[prize.equipment_path]")
+	updateUsrDialog()
+
+/obj/machinery/engi_points_manager/attackby(obj/item/I, mob/user, params)
+	return ..()
+
+/obj/machinery/engi_points_manager/process()
+	power_export_bonus = 0
+	for(var/obj/machinery/power/exporter/PE in machines)
+		power_export_bonus = PE.drain_rate/200
+	if(GBP_alarm_cooldown <= world.time)
+		var/limit = 0 // ugh, to stop it from checking the Centcom Computer
+		for(var/obj/machinery/computer/station_alert/SA in machines)
+			if(!limit)
+				air_alarm_bonus = max(0,(1000 - (SA.air_alarm_count * 200)))
+				power_alarm_bonus = max(0,(1000 - (SA.power_alarm_count * 200)))
+				fire_alarm_bonus = max(0,(500 - (SA.fire_alarm_count * 100)))
+				total_bonus = air_alarm_bonus + power_alarm_bonus + fire_alarm_bonus
+				limit++
+		switch(total_bonus)
+			if(0)
+				alarm_rating = "NOT WORTH THE AIR YOU'RE BREATHING, CONSIDER SUICIDE"
+				playsound(src, 'sound/misc/compiler-failure.ogg', 100, 1)
+			if(100 to 900)
+				alarm_rating = "COMPLICIT IN THE STATION'S DOWNFALL"
+				playsound(src, 'sound/misc/compiler-failure.ogg', 100, 1)
+			if(1000 to 1500)
+				alarm_rating = "HALF-ASSED"
+				playsound(src, 'sound/misc/compiler-stage1.ogg', 100, 1)
+			if(1600 to 2000)
+				alarm_rating = "ADEQUATE AND UNREMARKABLE"
+				playsound(src, 'sound/misc/compiler-stage1.ogg', 100, 1)
+			if(2100 to 2400)
+				alarm_rating = "IMPRESSIVE"
+				playsound(src, 'sound/misc/compiler-stage2.ogg', 100, 1)
+			if(2500)

```

Comment:
`total_bonus` isn't capped, thus can go above `2500`, this should be `=>` or (if that doesn't work, I don't remember)`to INF`